File: app/src/main/java/com/varunbarad/popularmovies/model/data/CollectionStub.java
package com.varunbarad.popularmovies.model.data;

import com.google.gson.Gson;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Creator: vbarad
 * Date: 2016-12-07
 * Project: PopularMovies
 */

public final class CollectionStub {
  @Expose
  @SerializedName("id")
  private long id;
  @Expose
  @SerializedName("name")
  private String name;
  @Expose
  @SerializedName("poster_path")
  private String posterPath;
  @Expose
  @SerializedName("backdrop_path")
  private String backdropPath;

  /**
   * Public no-args constructor for serialization
   */
  public CollectionStub() {

  }

  /**
   * @param id
   * @param name
   * @param posterPath
   * @param backdropPath
   */
  public CollectionStub(final long id, final String name, final String posterPath, final String backdropPath) {
    this.id = id;
    this.name = name;
    this.posterPath = posterPath;
    this.backdropPath = backdropPath;
  }

  public final long getId() {
    return id;
  }

  public final String getName() {
    return name;
  }

  public final String getPosterPath() {
    return posterPath;
  }

  public final String getBackdropPath() {
    return backdropPath;
  }

  @Override
  public String toString() {
    return (new Gson()).toJson(this);
  }
}

File: app/src/main/java/com/varunbarad/popularmovies/model/data/CollectionStub.kt
package com.varunbarad.popularmovies.model.data

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import com.squareup.moshi.Moshi

/**
 * Creator: Varun Barad
 * Date: 2019-06-05
 * Project: PopularMovies
 */
@JsonClass(generateAdapter = true)
data class CollectionStub(
    @Json(name = "id") val id: Long,
    @Json(name = "name") val name: String = "",
    @Json(name = "poster_path") val posterPath: String = "",
    @Json(name = "backdrop_path") val backdropPath: String = ""
) {
    override fun toString(): String {
        return Moshi.Builder().build().adapter(CollectionStub::class.java).toJson(this)
    }
}
